<div class='alert alert-warning'>

NumPy's interactive examples are experimental and may not always work as expected, with high load times especially on low-resource platforms, and the version of NumPy might not be in sync with the one you are browsing the documentation for. If you encounter any issues, please report them on the [NumPy issue tracker](https://github.com/numpy/numpy/issues).

</div>

An example from Glantz [1], pp 47-40:

Two groups, children of diabetics (25 people) and children from people
without diabetes (25 controls). Fasting blood glucose was measured,
case group had a mean value of 86.1, controls had a mean value of
82.2. Standard deviations were 2.09 and 2.49 respectively. Are these
data consistent with the null hypothesis that the parents diabetic
status does not affect their children's blood glucose levels?
Calculating the F statistic from the data gives a value of 36.01.

Draw samples from the distribution:


In [ ]:
dfnum = 1. # between group degrees of freedom
dfden = 48. # within groups degrees of freedom
rng = np.random.default_rng()
s = rng.f(dfnum, dfden, 1000)

The lower bound for the top 1% of the samples is :


In [ ]:
np.sort(s)[-10]

7.61988120985 # random

So there is about a 1% chance that the F statistic will exceed 7.62,
the measured value is 36, so the null hypothesis is rejected at the 1%
level.

The corresponding probability density function for ``n = 20``
and ``m = 20`` is:


In [ ]:
import matplotlib.pyplot as plt
from scipy import stats
dfnum, dfden, size = 20, 20, 10000
s = rng.f(dfnum=dfnum, dfden=dfden, size=size)
bins, density, _ = plt.hist(s, 30, density=True)
x = np.linspace(0, 5, 1000)
plt.plot(x, stats.f.pdf(x, dfnum, dfden))
plt.xlim([0, 5])
plt.show()